app/hooks/useMessageScroll.ts

```
@@ -1,11 +1,62 @@
+    scrollElement.scrollTop +
+      anchorRect.top -
+      scrollRect.top -
+      CHAT_TIMELINE_ANCHOR_OFFSET,
+  );
+}
+
+export const useMessageScroll = (anchorMessageId: string | null = null) => {
+  // use-stick-to-bottom retains the target callback created on mount, so the
+  // callback must read the current turn rather than close over its first ID.
+  const anchorMessageIdRef = useRef(anchorMessageId);
+  anchorMessageIdRef.current = anchorMessageId;
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/app/hooks/useMessageScroll.ts:49:3
  47 |   // callback must read the current turn rather than close over its first ID.
  48 |   const anchorMessageIdRef = useRef(anchorMessageId);
> 49 |   anchorMessageIdRef.current = anchorMessageId;
     |   ^^^^^^^^^^^^^^^^^^^^^^^^^^ Cannot update ref during render
  50 |
  51 |   const stickToBottom = useStickToBottom({
  52 |     resize: "smooth",

(react-hooks/refs)